app/scripts/modules/core/cluster/filter/clusterFilter.component.ts

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+import {CLUSTER_FILTER_SERVICE} from 'core/cluster/filter/clusterFilter.service';
+
+import {compact, uniq, map} from 'lodash';
+import {IScope, module} from 'angular';
+import {Application} from '../../application/application.model';
+
+export const CLUSTER_FILTER = 'cluster';
+
+let ngmodule = module(CLUSTER_FILTER, [
+  require('./collapsibleFilterSection.directive.js'),
+  CLUSTER_FILTER_SERVICE,
+  require('./clusterFilter.model.js'),
+  require('../../filterModel/dependentFilter/dependentFilter.service.js'),
+  require('./clusterDependentFilterHelper.service.js'),
+]);
+
+class ClusterFilterCtrl {
+  public categoryHeadings: any;
+  public stackHeadings: string[];
+  public instanceTypeHeadings: any;
+  public regionHeadings: any;
+  public accountHeadings: any;
+  public providerTypeHeadings: any;
+  public availabilityZoneHeadings: any;
```

Comment:
Can the `*Headings` be typed more strongly? Guessing they are `string[]`s but maybe `{[key: string]: boolean}[]`s?